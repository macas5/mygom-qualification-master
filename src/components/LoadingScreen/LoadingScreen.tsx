import {FC} from 'react';
import './loadingScreen-style.scss';

interface ILoadingScreen {
  isLoading: boolean;
}

const LoadingScreen: FC<ILoadingScreen> = ({isLoading}) => {
  if (isLoading) {
    return (
      <div className="lds-roller"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
    )
  } else
    return (null);
};

export default LoadingScreen;
