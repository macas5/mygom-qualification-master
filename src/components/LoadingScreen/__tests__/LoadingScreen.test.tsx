import React from "react";
import renderer from 'react-test-renderer';
import LoadingScreen from "../LoadingScreen";

it('renders correctly if loading', () => {
  const tree = renderer
    .create(<LoadingScreen isLoading={true}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly if not loading', () => {
  const tree = renderer
    .create(<LoadingScreen isLoading={false}/>).toJSON();
  expect(tree).toMatchSnapshot();
});