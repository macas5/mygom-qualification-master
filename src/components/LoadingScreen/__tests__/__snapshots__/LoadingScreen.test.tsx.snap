// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if loading 1`] = `
<div
  className="lds-roller"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`renders correctly if not loading 1`] = `null`;
